timelock-server/src/main/java/com/palantir/atlasdb/timelock/lock/LockAcquirer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.lock;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LockAcquirer {
+
+    private static final Logger log = LoggerFactory.getLogger(LockAcquirer.class);
+
+    public CompletableFuture<HeldLocks> acquireLocks(UUID requestId, List<AsyncLock> locks) {
+        CompletableFuture<Void> future = acquireLocksInOrder(requestId, locks, false);
+        registerErrorHandler(future, requestId, locks);
+
+        return future.thenApply(ignored -> new HeldLocks(locks, requestId));
+    }
+
+    public CompletableFuture<Void> waitForLocks(UUID requestId, List<AsyncLock> locks) {
+        return acquireLocksInOrder(requestId, locks, true);
+    }
+
+    private CompletableFuture<Void> acquireLocksInOrder(UUID requestId, List<AsyncLock> locks,
```

Comment:
Are we trusting the caller to put these locks in the right order?  Can we either verify the order here or convert `List<AsyncLock>` to `AsyncLocks` which enforces order.